### PR TITLE
[docs] Add notes about drift detection

### PR DESCRIPTION
**Pull request by t-kikuc**
**What this PR does / why we need it**:

1. ~~Fix:  DriftDetection is **DISABLED** by default.~~.  Add Note: DriftDetection is enabled by default, but automatic deployment is disabled by default.

2. Add Note:  `ignoreFields` is supported for only K8sApp.
   - Only used in here: https://github.com/pipe-cd/pipecd/blob/7ed87bcb1c0a4c603a24b78bb9ff89f98d54d0f1/pkg/app/piped/driftdetector/kubernetes/detector.go#L196

### CODE REVIEW

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 1]
```diff
@@ -807,7 +807,7 @@ Note: By default, the sum of traffic is rounded to 100. If both `primary` and `c
 
 | Field | Type | Description | Required |
 |-|-|-|-|
-| ignoreFields | []string | List of fields path in manifests, which its diff should be ignored. | No |
+| ignoreFields | []string | List of fields path in manifests, which its diff should be ignored. This is available for only `KubernetesApp`.This is available for only `KubernetesApp`. | No |
```
Comment (Warashi) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:810]: This is mistakenly repeated twice.

```suggestion
| ignoreFields | []string | List of fields path in manifests, which its diff should be ignored. This is available for only `KubernetesApp`. | No |
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ This status means the application is deploying and the configuration drift detec
 
 ### How to enable
 
-This feature is automatically enabled for all applications.
+By default, drift detection is **NOT** enabled. You can enable it for each application by configuring application's [OnOutOfSync.disabled](../configuration-reference/#onoutofsync).
```
Comment (Warashi) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:53]: The "drift detection" feature is the detection feature, not the reconcile feature.
The option you mention in this PR is the reconcile option; the deployment will be triggered when the sync state is OUT_OF_SYNC.

So, I mean, the drift detection is enabled by default.

Comment (khanhtc1202) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:53]: Yes, the drift detection is enabled by default. The disabled code is disabled the auto trigger new deployment on OUT_OF_SYNC by default @t-kikuc

File: docs/content/en/docs-dev/user-guide/configuration-reference.md (hunk 1 above)
Comment (t-kikuc) [reply docs/content/en/docs-dev/user-guide/configuration-reference.md]: thanks, I fixed

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md (hunk 2 above)
Comment (t-kikuc) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md]: thanks both, 
yeah, drift detection is enabled by default, and deployment on OutOfSync is not.

I recovered the previous sentence and added a new explanation. [2f0785b](https://github.com/pipe-cd/pipecd/pull/5120/commits/2f0785b8dc0082e8c6c5339d602d0b32cf39fb4b)

```
This feature is automatically enabled for all applications.

If you want to deploy automatically when `OUT_OF_SYNC` occures, configure each application's [OnOutOfSync](../configuration-reference/#onoutofsync).
```

**Review by Warashi — DISMISSED** [review]
LGTM

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md (hunk 2 above)
Comment (khanhtc1202) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md]: @t-kikuc There is a long-time story about this spec 😓 
ref: https://github.com/pipe-cd/pipecd/discussions/2752
tl;dr: Previously, we didn't support trigger deployment on OUT_OF_SYNC, which someone thinks that it didn't suit the GitOps spec, so we decided to add that feature. The problem is, if we make that trigger by default, our users at that time will be affected (due to a ton of unexpected deployment triggers whenever OUT_OF_SYNC happens). That's why we make this spec - disable trigger deployment on OUT_OF_SYNC by default. 😄

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 3]
```diff
@@ -52,6 +52,8 @@ This status means the application is deploying and the configuration drift detec
 
 This feature is automatically enabled for all applications.
 
+If you want to deploy automatically when `OUT_OF_SYNC` occures, configure each application's [OnOutOfSync](../configuration-reference/#onoutofsync).
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:55]:
```suggestion
If you want to trigger deployment automatically when `OUT_OF_SYNC` occurs, configure each application's [OnOutOfSync](../configuration-reference/#onoutofsync).
```
typo and re-write 👀

Comment (khanhtc1202) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:55]: Same for the others

Comment (khanhtc1202) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:55]: On the second thought, I think this note could lead to confusing situation 🤔 We have place explain how to trigger pipecd deployment here: https://pipecd.dev/docs-v0.48.x/user-guide/managing-application/triggering-a-deployment/#trigger-configuration
Merging this note here could lead to confusion between drift detection feature and trigger feature of pipecd. WDYT? @t-kikuc @Warashi

Comment (Warashi) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:55]: This document is not a document like a dictionary, so we can write the same thing on several pages.
However, we can avoid confusion by the guide having centralized, with only one page at one thing.
I propose that this notation have a link to the page @khanhtc1202 mentioned, not the configuration reference page.

Comment (t-kikuc) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:55]: >I propose that this notation have a link to the page @khanhtc1202 mentioned, not the configuration reference page.

I agree

Comment (t-kikuc) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md]: I fixed: [a25e438](https://github.com/pipe-cd/pipecd/pull/5120/commits/a25e43839290cef322b4d39cac548c5435850ba9)

```markdown
### How to enable

This feature is automatically enabled for all applications.

If you want to trigger deployment automatically when `OUT_OF_SYNC` occurs, see [Trigger configuration](./triggering-a-deployment/#trigger-configuration).
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 4]
```diff
@@ -52,6 +52,8 @@ This status means the application is deploying and the configuration drift detec
 
 This feature is automatically enabled for all applications.
 
+If you want to trigger deployment automatically when `OUT_OF_SYNC` occurs, see [Trigger configuration](./triggering-a-deployment/#trigger-configuration).
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:55]: This should be a note and place under L57
```
This feature is automatically enabled for all applications.

You can change the checking interval as well as [configure the notification](../../managing-piped/configuring-notifications/) for these events in `piped` configuration.

Note: If you want to trigger deployment automatically when `OUT_OF_SYNC` occurs, see [Trigger configuration](./triggering-a-deployment/#trigger-configuration).
```

Comment (t-kikuc) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md]: thanks, i fixed [7b6dba3](https://github.com/pipe-cd/pipecd/pull/5120/commits/7b6dba380c85d6b8e0afa03b2c3806b521bcc67e)

**Review by khanhtc1202 — APPROVED** [review]
Thanks

**Review by Warashi — APPROVED** [review]
thanks
